lib/publiccloud/instance.pm

```diff
@@ -255,19 +260,26 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
-=head2 wait_for_guestregister
+=head2 wait_for_guestregister_chk
```

Comment:
This does not belong here.

Reply:
Yes sorry, I forgot to clarify it, we discussed this change, during our meeting last Mar 10.

For this poo 118027 about boottime measure triggered in `provider::create_instances`, we need the _boot finished_ and _guest register checked_ before measurement start.

Given that in almost all modules where `create_instances` used it is soon  followed by a call to `wait_for_guestregister`, we agreed to move this last function inside of `it`.

Now as `first phase,` to soon verify the new internal location and reduce the changes of deleting all external 'wait_for_guestregister' calls, I "renamed" it for the internal call as `wait_for_guestregister_chk` and the _old_ `wait_for_guestregister`  changed to perform a simple display data where still present as external call.

Reply:
Also  I added to `create_instance` a more input paramenter `check_guestregister`, to enable-1/disable-0 that  _internal_ function call where not needed, like in the 4 modules here also updated with `create_instance(check_guestregister => 0)`

Reply:
nevertheless you edited docstring which still belongs to `wait_for_guestregister` so you need to move this documentation below `wait_for_guestregister_chk`  . I assume Pavel meant that

Reply:
yes, will do it